bigtable/bigtable.go

```diff
@@ -113,15 +114,40 @@ var (
 	isIdempotentRetryCode = make(map[codes.Code]bool)
 	retryOptions          = []gax.CallOption{
 		gax.WithRetry(func() gax.Retryer {
-			return gax.OnCodes(idempotentRetryCodes, gax.Backoff{
+			backoff := gax.Backoff{
 				Initial:    100 * time.Millisecond,
 				Max:        2 * time.Second,
 				Multiplier: 1.2,
-			})
+			}
+			return &bigtableRetryer{
+				Retryer: gax.OnCodes(idempotentRetryCodes, backoff),
+				Backoff: backoff,
+			}
 		}),
 	}
 )
 
+// bigtableRetryer extends the generic gax Retryer, but also checks
+// error messages to check if operation can be retried
+type bigtableRetryer struct {
+	gax.Retryer
+	gax.Backoff
+}
+
+func (r *bigtableRetryer) Retry(err error) (time.Duration, bool) {
+	if status.Code(err) == codes.Internal &&
+		strings.Contains(err.Error(), "stream terminated by RST_STREAM") {
```

Comment:
is there a way to make this more flexible? Like take in a list or regex string describing the retryable error messages?